mage_ai/services/k8s/job_manager.py

```diff
@@ -83,60 +86,67 @@ def run_job(
             raise Exception(f'Failed to execute k8s job {self.job_name}')
 
     def create_job_object(
-        self,
-        command: str,
-        k8s_config: K8sExecutorConfig = None,
+            self,
+            command: str,
+            k8s_config: K8sExecutorConfig = None,
     ) -> client.V1Job:
-        # Configureate Pod template container
-        mage_server_container_spec = self.pod_config.spec.containers[0]
-
-        container_kwargs = dict(
-            name='mage-job-container',
-            image=mage_server_container_spec.image,
-            image_pull_policy='IfNotPresent',
-            command=command.split(' ') if isinstance(command, str) else command,
-            env=mage_server_container_spec.env,
-            volume_mounts=mage_server_container_spec.volume_mounts,
-        )
-        if k8s_config and (k8s_config.resource_limits or k8s_config.resource_requests):
```

Comment:
Again, your change here is not backward compatible. It'll break existing usage of the k8s executor.

Reply:
I added a function to merge containers and keep backward compatibility

Reply:
Hmm, these params in the  are still ignored.
```
    namespace: str = DEFAULT_NAMESPACE
    resource_limits: Dict = None
    resource_requests: Dict = None
    service_account_name: str = DEFAULT_SERVICE_ACCOUNT_NAME
```
https://docs.mage.ai/production/configuring-production-settings/compute-resource#kubernetes-executor
Could you add unit tests for different config to make sure it's backward compatible?

Reply:
I made some changes to maintain the same old logic for those parameters. By the way, maybe we need to propagate the changes and enforce the use of a structured entity of exchange in K8S related classes.

Reply:
Thank you!